cppcache/src/PdxTypeRegistry.hpp

```diff
@@ -53,7 +53,20 @@ typedef std::unordered_map<std::shared_ptr<PdxSerializable>,
                            dereference_hash<std::shared_ptr<CacheableKey>>,
                            dereference_equal_to<std::shared_ptr<CacheableKey>>>
     PreservedHashMap;
-typedef std::map<std::shared_ptr<PdxType>, int32_t, PdxTypeLessThan>
+
+struct PdxTypeHashCode {
+  std::size_t operator()(std::shared_ptr<PdxType> const& pdx) const {
+    return pdx ? pdx->hashcode() : 0;
+  }
+};
+
+struct PdxTypeEqualCmp {
+  bool operator()(std::shared_ptr<PdxType> const& first, std::shared_ptr<PdxType> const& second) const{
+    return first->hashcode() == second->hashcode();
```

Comment:
Equality of hashcode does not guarantee equality of the object. You should just implement `PdxType::operator==`.

Reply:
Agree, but I dont want to check if the objects are equals, I just want to check if two `PdxType` objects have the same hash.

Two `PdxType` objects should be equals if they have the same classname and their fields have the same types, names and values.
And two keys of the `pdxTypeToTypeId` map will be equals if they have the same hash, which means they have the same classname, and their fields have the same types and names (values are not checked here).

Reply:
Sorry, I read again, I got your point, two objects could generate the same hash being different :)